dotCMS/src/main/java/com/dotcms/security/secret/ServiceDescriptor.java

```diff
@@ -0,0 +1,81 @@
+package com.dotcms.security.secret;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class ServiceDescriptor {
+
+    private String key;
```

Comment:
it seems these attributes could be final

Reply:
nope, since I have to have an empty constructor they can not be made final 
or the objectMapper that reads them will fail

Reply:
I see, Builder?